src/lib/io/sol-efivarfs-storage.c

```diff
@@ -214,11 +214,7 @@ sol_efivars_read_raw(const char *name, struct sol_buffer *buffer)
         goto end;
     }
 
-    if (buffer->capacity) {
-        r = sol_util_fill_buffer_exactly(fd, buffer, buffer->capacity);
-    } else {
-        r = sol_util_load_file_fd_buffer(fd, buffer);
-    }
+    r = sol_util_load_file_fd_buffer(fd, buffer);
```

Comment:
why is this change? I don't recall the details, but I remember once I tried to do this I ran into problems. Did you try with both fixed capacity and growing buffers?